server/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java

```diff
@@ -817,6 +830,30 @@ public Settings archiveUnknownOrInvalidSettings(
         }
     }
 
+    public Settings removeUnknownSettings(Settings settings) {
+        Settings.Builder builder = Settings.builder();
+        boolean changed = false;
+        for (final String key : settings.keySet()) {
+            final Setting<?> setting = getRaw(key);
+            if (((isPrivateSetting(key) || (setting != null && setting.isPrivateIndex())))) {
+                continue;
+            }
+            if (key.startsWith(ARCHIVED_SETTINGS_PREFIX)) {
+                continue;
+            }
+            if (setting != null) {
+                builder.copy(key, settings);
+            } else {
+                // Builder doesn't include this, actual difference.
+                changed = true;
+            }
```

Comment:
If I read this correctly, then a changed copy won't include archived or private settings, but an unchanged copy will.  That sounds a bit confusing to me?

Reply:
Good catch, thanks!
Archived and private settings must stay, I pushed a fix and updated a test to catch this.